### ISSUE TITLE
DataClassContainsFunctions - allow conversion function

### ISSUE THREAD

**Issue by vanniktech** [arturbosch/detekt#434]
What do you think about an option that allows conversion functions that take the Data class and basically map it to a different type. For instance:

```kotlin
data class MySomething(val prop1: String, ...) {
  fun toSomething() = Something(prop1)
}
```

**Comment by schalkms** [comment]
Is this related to the `UseDataClass` rule?
Does that mean that a normal class which uses conversion functions should also be a data class candidate?

**Comment by vanniktech** [comment]
Hmm it might yeah. I'd say that conversion functions are defined with a prefix of `to` or `as` and then no arguments.

**Comment by arturbosch** [comment]
Actually I also often use conversion functions for data classes but outside the class and in the same file. I would also argue that this should be the preferred way.

**Comment by tagantroy** [comment]
@vanniktech @arturbosch Hello, should I add this feature to existing rules by default ( DataClassContainsFunctions, UseDataClass) or add additional configuration?

**Comment by arturbosch** [comment]
@tagantroy I don't know. I still think such functions should be put at top level (extension functions). If you guys ( @vanniktech @schalkms ) give the issue a thumbs up, then why not @tagantroy  :)

**Comment by schalkms** [comment]
I agree with @arturbosch
I wouldn't add conversion functions to data classes. However, I also think an optional configuration is ok if it's set to **false** per default.

**Comment by arturbosch** [comment]
@tagantroy here you have your go :) thanks for contributing!

**Comment by tagantroy** [comment]
@arturbosch already merged, can we close it?

**Comment by Mauin** [comment]
Closed via #445
